android/src/main/java/com/bindidreactnative/TSBindIdParser.java

```diff
@@ -57,9 +58,19 @@ public XmBindIdConfig getConfig(Context applicationContext, ReadableMap map) {
     String environmentUrl = environment.getString("environmentUrl");
     XmBindIdServerEnvironmentMode environmentMode = TSBindIdUtils.getEnum(environment, "environmentMode", XmBindIdServerEnvironmentMode.class);
 
-    XmBindIdServerEnvironment serverEnvironment = XmBindIdServerEnvironment.createWithMode(environmentMode);
-    serverEnvironment.setEnvironmentUrl(environmentUrl);
+    XmBindIdServerEnvironment serverEnvironment;
 
+    if (environmentUrl != null && !environmentUrl.isEmpty()) {
+       serverEnvironment = XmBindIdServerEnvironment.createWithUrl(environmentUrl);
+    } else {
+       serverEnvironment = XmBindIdServerEnvironment.createWithMode(environmentMode);
+    }
+
+    if (environment.hasKey("tokenExchangePlatformMode")) {
```

Comment:
Since `tokenExchangePlatformMode` is being used multiple times it should be referenced from a constant key.